.github/workflows/release.yml

```diff
@@ -0,0 +1,236 @@
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          role-to-assume: ${{ secrets.AWS_ASSUME_ROLE_ARN_PROD }}
+          role-duration-seconds: 1200
+          aws-region: ${{ matrix.aws_region }}
+          mask-aws-account-id: false
+      - name: publish
+        run: |
+          BUCKET_NAME=${{ github.event.inputs.toLayerName }}-${{ github.run_id }}-${{ matrix.aws_region }}
+          aws s3 mb s3://$BUCKET_NAME
+          aws s3 cp layer.zip s3://$BUCKET_NAME
+          layerARN=$(aws lambda publish-layer-version --layer-name ${{ github.event.inputs.toLayerName }} --content S3Bucket=$BUCKET_NAME,S3Key=layer.zip --query 'LayerVersionArn' --output text)
+          echo $layerARN
+          mkdir ${{ github.event.inputs.toLayerName }}
+          echo $layerARN > ${{ github.event.inputs.toLayerName }}/${{ matrix.aws_region }}
+          cat ${{ github.event.inputs.toLayerName }}/${{ matrix.aws_region }}
+      - name: public layer
+        run: |
+          layerVersion=$(aws lambda list-layer-versions --layer-name ${{ github.event.inputs.toLayerName }} --query 'max_by(LayerVersions, &Version).Version')
+          aws lambda add-layer-version-permission --layer-name ${{ github.event.inputs.toLayerName }} --version-number $layerVersion --principal "*" --statement-id publish --action lambda:GetLayerVersion
+      - name: upload layer arn artifact
+        if: ${{ success() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ github.event.inputs.toLayerName }}
+          path: ${{ github.event.inputs.toLayerName }}/${{matrix.aws_region}}
+      - name: clean s3
+        if: always()
+        run: |
+          BUCKET_NAME=${{ github.event.inputs.toLayerName }}-${{ github.run_id }}-${{ matrix.aws_region }}
+          aws s3 rb --force s3://$BUCKET_NAME
+  update-code:
+    runs-on: publish-prod
+    needs: init
+    steps:
+      - name: download layerARNs
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ github.event.inputs.toLayerName }}
+          path: ${{ github.event.inputs.toLayerName }}
+      - name: show layerARNs
+        run: |
+          for file in ${{ github.event.inputs.toLayerName }}/*
+          do
+          echo $file
+          cat $file
+          done
+      - name: generate layer-note
+        run: |
+          echo "| Region | Layer ARN |" >> layer-note
+          echo "|  ----  | ----  |" >> layer-note
+          cd ${{ github.event.inputs.toLayerName }}
+          for file in *
+          do
+          read arn < $file
+          echo "| " $file " | " $arn " |" >> ../layer-note
+          done
+          cd ..
+          cat layer-note
+      # TODO: build splitting ARN TF
+      - name: build splitting TF file
+        run: echo 'todo'
+  smoke-test:
+    runs-on: ubuntu-latest
+    needs: [ init, update-code ]
+    strategy:
+      matrix:
```

Comment:
Fix me if there is better idea

Reply:
Since this is run after publishing, can we just workflow trigger the canary workflow?

Reply:
the smoke test is similar to canary, I copied most of code from canary.yml. But cannot reuse current canary workflow. The differences:
1. Release only process one layer kind(python, nodejs, javawrapper, javaagent) at one time. don't have to run all canary sample apps.
2. Have to update layer to new releasing layer, by overriding splitting ARN tf file.

Reply:
Since we may have no enough time to finish smoke test automation in release workflow, can we manually do release by steps:
1. run `release` workflow, get layers in prod.
2. manually submit PR for splitting ARN tf files for canary sample applications.
3. trigger canary test workflow, make sure canary success.
4. update README.md with layer ARNs.